[fix] deploy.sh 안정화 및 promtail-config 생성 방식 개선

zerone1202 opened the pull request:
## #️⃣ Issue Number<!--- ex) #이슈번호, #이슈번호 --> 

- Closes #395

## 📝 요약(Summary)<!--- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->

- heredoc 방식으로 promtail-config.yml 생성 안정화
- Redis / App / Promtail 실행 순서 보정
- SSM 명령 타임아웃 발생 가능성 감소
- 배포 로그 및 파라미터 로딩 흐름 정리

## 🛠️ PR 유형

어떤 변경 사항이 있나요?

- [ ] 새로운 기능 추가
- [x] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [x] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## ✅ PR Checklist

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.

zerone1202 committed:
[fix] common: deploy.sh 안정화 및 promtail 설정 파일 생성 방식 개선

- heredoc 방식으로 promtail-config.yml 생성 안정화
- Redis / App / Promtail 실행 순서 보정
- SSM 명령 타임아웃 발생 가능성 감소
- 배포 로그 및 파라미터 로딩 흐름 정리

zerone1202 commented on scripts/deploy.sh:
```diff
@@ -17,81 +17,92 @@ REPO_AND_TAG="$(echo "${FULL_URI}" | cut -d/ -f2- )"
+  "echo '[INFO] Fetching Redis config from Parameter Store...'"
   "export REDIS_HOST=\$(aws ssm get-parameter --name '/config/dev/redis.host' --query 'Parameter.Value' --output text --region ${AWS_REGION})"
   "export REDIS_PORT=\$(aws ssm get-parameter --name '/config/dev/redis.port' --query 'Parameter.Value' --output text --region ${AWS_REGION})"
   "export REDIS_PASSWORD=\$(aws ssm get-parameter --name '/config/dev/redis.password' --with-decryption --query 'Parameter.Value' --output text --region ${AWS_REGION})"
```
반영하겠습니다.

DoH100 reviewed (commented):
고생하신게 눈에 보입니다.. 고생하셨습니다..

zerone1202 committed:
[fix] common: SSM Redis 파라미터 조회를 단일 get-parameters 호출로 통합

- Redis 설정 조회 과정의 API 호출을 3회 → 1회로 감소
- SSM 네트워크 지연 및 타임아웃 발생 가능성 완화
- 배포 스크립트 안정성 및 실행 속도 개선

assokk reviewed (approved):
확인했습니다 고생하셨어요! :D
